src/orion/core/utils/random_state.py

```diff
@@ -80,26 +81,26 @@ def seed(cls, base_seed: int | None) -> RandomState:
         return random_state
 
 
-class Algo(Protocol):
-    """Protocol for algorithms with a random state that can be saved and restored."""
+class HasRandomState(Protocol):  # pylint: disable=too-few-public-methods
+    """Protocol for objects with a random state that can be saved and restored."""
 
     random_state: RandomState | None
 
 
 @contextmanager
-def control_randomness(algo: Algo):
+def control_randomness(has_random_state: HasRandomState):
     """Seeds the randomness inside the indented block of code using `self.random_state`."""
-    if algo.random_state is None:
+    if has_random_state.random_state is None:
```

Comment:
One last thing. Wouldn't we want to set the property, if the object doesn't already have one? (for example, by just doing the same thing as below, but without the initial set()?
Or do we want to keep that object completely unseeded, if its `random_state` attribute is None?

Reply:
I'd say keep the object completely unseeded.